doc/user_guide/ensembleModel.tex

```diff
@@ -0,0 +1,229 @@
+models is run for each sample taken by RAVEN.  If instead you want to postprocess a batch of sampled data, try
+the PostProcessor models instead.
+
+\subsection{Introduction: The Ensemble Model}
+The key to the Ensemble Model is careful definition of the inputs and outputs of each model.  When an
+Ensemble Model is defined in a RAVEN run, RAVEN will automatically scan the inputs and outputs of each model
+and determine the right order to evaluate the models in (or \emph{graph}).
+
+\subsection{Example: ballistics and impact}
+% TODO add an image showing these two models, their inputs and outputs
+As a basic example, consider two physics models.  The first is a ballistics code, used to determine the
+kinetic energy $E$ of a ball when it hits the ground, with a path depending on its initial height $y_0$, initial velocity
+$v_0$, mass $m$ and initial angle $\theta_0$.  We'll assume we're near the Earth's surface and aside from the initial
+launch height, we're launching over a flat surface.  This can be represented as a functional $E(y_0,v_0,m,\theta_0)$.
+
+The second physics model is an impact code that estimates the diameter of the crater $D$ made when a ball
+impacts.  With a few simplifying assumptions ($D$ is under 1 km, the impact is near Earth's surface, the
+impact is in dry sand with fixed density), size $D$ can be determined as a function of the ball's kinetic
+energy when impacting $E$, its mass $m$, and its radius $r$: $D(E,m,r)$.
+
+We can see that the input $E$ for the impact model is calculated by the ballistics code.  In RAVEN the
+ensemble model will automatically detect this, and run the ballistics code before the impact code in each
+iteration.
+
+In general, the EnsembleModel also has some finite capacity for resolving circular dependencies using Picard
+iteration.  See the manual for more information on this capability.
+
+Setting up an ensemble model can be a tricky process, so we'll walk through the input using our example
+outlined above.  For
+our purposes, we're going to let RAVEN perturb the codes using the GenericCode interface.  We'll call the
+ballistics code \texttt{ballistics.py} with keyword input file \texttt{ballistics\_input.txt} and similarly the impact
+code \texttt{impact.py} with keyword input file \texttt{impact\_input.txt}.
+
+The RAVEN input file for this example is located at
+\begin{verbatim}
+raven/tests/framework/user_guide/EnsembleModel/basic.xml
+\end{verbatim}
+with run directory \texttt{run\_basic}.  The models and inputs are in the run directory.
+
+Now we turn our attention to the RAVEN input file. We will discuss \xmlNode{DataObjects}, \xmlNode{Files},
+\xmlNode{Models}, and \xmlNode{Steps} in detail; the rest are typical usage and need no special attention.
+
+\subsubsection{DataObjects}
+The key to a successful EnsembleModel is setting up the data object inputs and outputs.  Each model has a
```

Comment:
I would replace ```data object ``` with ```DataObject``` just to keep a "mental" link between the writing and the input

Reply:
Does the mental link establishment outweigh how jarring it is to read?